BigBaseV2/src/views/vehicle/view_lsc.cpp

```diff
@@ -221,35 +244,78 @@ namespace big
 
 					if (ImGui::Selectable(name.c_str(), mod == owned_mods[selected_slot]))
 					{
+						owned_mods[selected_slot] = mod;
+
 						g_fiber_pool->queue_job([&mod] {
 							NETWORK::NETWORK_REQUEST_CONTROL_OF_ENTITY(self::veh);
 							if (selected_slot >= 0)
 							{
 								VEHICLE::SET_VEHICLE_MOD(player_vehicle, selected_slot, mod, false);
-								owned_mods[selected_slot] = VEHICLE::GET_VEHICLE_MOD(player_vehicle, selected_slot);
 							}
 							else if (selected_slot == MOD_WINDOW_TINT)
 							{
 								VEHICLE::SET_VEHICLE_WINDOW_TINT(player_vehicle, mod);
-								owned_mods[selected_slot] = VEHICLE::GET_VEHICLE_WINDOW_TINT(player_vehicle);
 							}
 							else if (selected_slot == MOD_WHEEL_TYPE)
 							{
 								VEHICLE::SET_VEHICLE_WHEEL_TYPE(player_vehicle, mod);
 								VEHICLE::SET_VEHICLE_MOD(player_vehicle, MOD_FRONTWHEEL, 0, false);
-								owned_mods[selected_slot] = VEHICLE::GET_VEHICLE_WHEEL_TYPE(player_vehicle);
-								owned_mods[MOD_FRONTWHEEL] = VEHICLE::GET_VEHICLE_MOD(player_vehicle, MOD_FRONTWHEEL);
+								owned_mods[MOD_FRONTWHEEL] = 0;
 							}
 							else if (selected_slot == MOD_PLATE_STYLE)
 							{
 								VEHICLE::SET_VEHICLE_NUMBER_PLATE_TEXT_INDEX(player_vehicle, mod);
-								owned_mods[selected_slot] = VEHICLE::GET_VEHICLE_NUMBER_PLATE_TEXT_INDEX(player_vehicle);
 							}
 						});
 					}
 				}
 				ImGui::ListBoxFooter();
 			}
+
+			if (
+				selected_slot == MOD_FRONTWHEEL ||
+				selected_slot == MOD_REARWHEEL
+			) {
+				auto wheel_map = front_wheel_map;
+				auto wheel_mod = front_wheel_mod;
+
+				if (selected_slot == MOD_REARWHEEL)
+				{
+					wheel_map = rear_wheel_map;
+					wheel_mod = rear_wheel_mod;
+				}
+
+
+				ImGui::SameLine();
+				if (ImGui::ListBoxHeader("Style", ImVec2(200, 200)))
+				{
+					std::string mod_name = mod_display_names[selected_slot][owned_mods[selected_slot]];
+					auto wheel_mods = wheel_map[mod_name];
+
+					for (int i = 0; i < wheel_mods.size(); i++)
+					{
+						auto mod = wheel_mods[i];
+
+						std::string label = "Style " + std::to_string(i);
+
+						if (ImGui::Selectable(label.c_str(), mod == wheel_mod))
+						{
+							VEHICLE::SET_VEHICLE_MOD(player_vehicle, selected_slot, mod, true);
```

Comment:
Calling natives outside of script threads is detected.

Use the fiber pool please.